boltdesignsystem/bolt
# New Typeahead Options: Disable Filtering / Sorting, Render Custom Footer UI, Full Theming Support

## Jira
http://vjira2:8080/browse/BDS-1864

## Summary
Adds the ability to disable filtering and/or disable sorting Typeahead search results in addition to being able to customize how to map up complex search data + customize how these complex results get rendered / selected.

![image](https://user-images.githubusercontent.com/1617209/68808032-f19fbe00-0636-11ea-9fa8-6c40e1480a0c.png)

Furthermore, these updates also add 1st class support for color theming + being able to override the color theme via the new `theme` prop.

## Details
With these updates (code. demo, and doc updates), Typeahead now supports
- Customizing / Overriding Theming via the new `theme` prop
- Disabling Filtering / Sorting via the new `no-sort` and `no-filter` props
- Rendering custom footer content via the new `footerTemplate` function

Additionally while putting together the updated component demos, I tossed together a new `<bolt-component-demo>` helper to make it easy to switch between different themes when testing / demoing components.

## How to test
- Review code and docs changes
- Double check our support browsers are working as expected
- @christophersmith262 or @relimck especially, verify that these new APIs handle the downstream needs by the team
